ftests/036-cgset-multi_thread.py

```diff
@@ -0,0 +1,138 @@
+#!/usr/bin/env python3
+#
+# Basic cgrules functionality test
+#
+# Copyright (c) 2022 Oracle and/or its affiliates.
+# Author: Kamalesh Babulal <kamalesh.babulal@oracle.com>
+#
+
+#
+# This library is free software; you can redistribute it and/or modify it
+# under the terms of version 2.1 of the GNU Lesser General Public License as
+# published by the Free Software Foundation.
+#
+# This library is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU Lesser General Public License
+# for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with this library; if not, see <http://www.gnu.org/licenses>.
+#
+
+from cgroup import Cgroup, CgroupVersion
+from process import Process
+from run import Run
+import consts
+import ftests
+import sys
+import os
+
+CONTROLLER='cpu'
+PARENT_CGNAME='036cgset'
+CHILD_CGNAME='childcg'
+
+SETTING = 'cgroup.type'
+AFTER = 'threaded'
+THREADS = 3
+
+def prereqs(config):
+    result = consts.TEST_PASSED
+    cause = None
+
+    if config.args.container:
```

Comment:
Curiosity killed the cat, so I ran the test in a container.  It looks like it hung - I'm guessing on the deletion of the pids.  I saw this before on the cgexec test (34, I think).

I think we have to leave the container check in there for now.  Oh well.

Reply:
I thought about it, one way idea to containerize this would be to create a simple multithreaded python script as part of the `tests/ftest` module and call `lxc exec TestLibcg exec -- python3 multithread.py` instead of the current logic, that works under no container.